src/commands/util/Ping.ts

```diff
@@ -1,17 +1,226 @@
-import { CommandInteraction } from "discord.js";
+import { ApplicationCommandOptionType, Collection, CommandInteraction, EmbedBuilder, EmbedField } from "discord.js";
+import { isIP } from "net";
 import { Command } from "@src/Command";
+import { axios, embedify } from "@src/utils";
+import { settings } from "@src/settings";
+import { AxiosError } from "axios";
+
+interface MimeInfo {
+    exts: string[];
+    name: string;
+}
 
 export class Ping extends Command {
     constructor()
     {
         super({
             name: "ping",
-            desc: "Check if the bot is alive",
-            category: "util"
+            desc: "Visits a URL, follows redirects and gives you all kinds of information about it",
+            category: "util",
+            args: [
+                {
+                    name: "url",
+                    desc: "The URL to gather information about. Has to start with https:// or http://",
+                    type: ApplicationCommandOptionType.String,
+                    required: true,
+                },
+            ],
+            // TODO: implement proxy so we don't dox our IP lol
+            devOnly: true,
         });
     }
 
-    async run(int: CommandInteraction): Promise<void> {
-        await this.reply(int, "Pong!", true);
+    private readonly HOST_BLACKLIST = [ "localhost", "[::1]" ];
+    private readonly MIME_MAP = new Collection<string, MimeInfo>([
```

Comment:
maybe a constants.ts file ?? but idk if that's worth it. maybe we already have a file with lots of constants.
idk. @canarado what you think?

Reply:
This is fine since these objects are only accessed in one class, could be a refactor in the future if they are needed elsewhere